utils/buffer/examples/basic_buffer_pool.rs

```diff
@@ -0,0 +1,41 @@
+// # Simple `BufferPool` Usage
+//
+// This example showcases how to:
+// 1. Creating a `BufferPool`.
+// 2. Obtaining a writable buffer.
+// 3. Writing data into the buffer.
+// 4. Retrieving the data as a referenced slice.
+// 5. Retrieving the data as an owned slice.
+//
+// # Run
+//
+// ```
+// cargo run --example basic_buffer_pool
+// ```
+
+use buffer_sv2::{Buffer, BufferPool};
+
+fn main() {
+    // Create a new BufferPool with a capacity of 32 bytes
+    let mut buffer_pool = BufferPool::new(32);
+
+    // Get a writable buffer from the pool
+    let data_to_write = b"Ciao, mundo!"; // 12 bytes
+    let writable = buffer_pool.get_writable(data_to_write.len());
+
+    // Write data (12 bytes) into the buffer. Fills the first slot with 12 bytes, leaving 7
+    // remaining slots open to fit a maximum of 20 more bytes worth of data for future writes
+    writable.copy_from_slice(data_to_write);
+    assert_eq!(buffer_pool.len(), 12);
+
+    // Retrieve the data as a referenced slice
+    let _data_slice = buffer_pool.get_data_by_ref(12);
+    assert_eq!(buffer_pool.len(), 12); // Buffer still owns data
+
+    // Retrieve the data as an owned slice
+    let data_slice = buffer_pool.get_data_owned();
+    assert_eq!(buffer_pool.len(), 0); // Buffer is now empty
+
+    let expect_data = [67, 105, 97, 111, 44, 32, 109, 117, 110, 100, 111, 33];
+    assert_eq!(data_slice.as_ref(), expect_data);
```

Comment:
I switched this and the test still works... I feel `expect_data` is actually causing more confusion with no real benefit
```suggestion
    assert_eq!(data_slice.as_ref(), data_to_write);
```